### ISSUE TITLE
StackOverflowError for nested Enum

### ISSUE THREAD

**Issue by GoogleCodeExporter** [vadbch/orika#41]
```
What steps will reproduce the problem?

TypeFactory.valueOf some nested Enum throw StackOverflowError
due to an evantual bug in handling recursive Generics




```

Original issue reported on code.google.com by `elaat...@gmail.com` on 22 Aug 2012 at 8:56

**Comment by GoogleCodeExporter** [comment]
```
Applied fix from current trunk to the 1.2.0 branch
```

Original comment by `matt.deb...@gmail.com` on 23 Aug 2012 at 5:22

**Comment by GoogleCodeExporter** [comment]
Original comment by `elaat...@gmail.com` on 24 Aug 2012 at 10:14
- Changed state: **Fixed**
